src/Mod/Sketcher/App/planegcs/Constraints.cpp

```diff
@@ -2845,4 +2845,86 @@ double ConstraintC2CDistance::grad(double *param)
     return deriv * scale;
 }
 
+// --------------------------------------------------------
+// ConstraintC2LDistance
+ConstraintC2LDistance::ConstraintC2LDistance(Circle &c, Line &l, double *d)
+{
+    this->d = d;
+    pvec.push_back(d);
+
+    this->circle = c;
+    this->circle.PushOwnParams(pvec);
+
+    this->line = l;
+    this->line.PushOwnParams(pvec);
+
+    origpvec = pvec;
+    pvecChangedFlag = true;
+    rescale();
+}
+
+ConstraintType ConstraintC2LDistance::getTypeId()
+{
+    return C2LDistance;
+}
+
+void ConstraintC2LDistance::rescale(double coef)
+{
+    scale = coef;
+}
+
+void ConstraintC2LDistance::ReconstructGeomPointers()
+{
+    int i = 0;
+    i++;// skip the first parameter as there is the inline function distance for it
+    circle.ReconstructOnNewPvec(pvec, i);
+    line.ReconstructOnNewPvec(pvec, i);
+    pvecChangedFlag = false;
+}
+
+void ConstraintC2LDistance::errorgrad(double *err, double *grad, double *param)
+{
+    DeriVector2 ct (circle.center, param);
+    DeriVector2 p1 (line.p1, param);
+    DeriVector2 p2 (line.p2, param);
+    DeriVector2 v_line = p2.subtr(p1);
+
+    double u = -ct.x * v_line.y + ct.y * v_line.x + p1.x * p2.y - p2.x * p1.y; //center to line distance
+    double du = -ct.dx*v_line.y - ct.x * v_line.dy + ct.dy * v_line.dx + ct.y * v_line.dx
+                + p1.dx * p2.y + p1.x * p2.dy  - p2.dx * p1.y - p2.x * p1.dy;
+
+    double dv;
+    double v = v_line.length(dv);
+    double h = u / v;
+    double dh = (du * v - u * dv) / (v * v);
+
```

Comment:
Please, try to use better names for the variables and/or put some comments about what exactly is each. Also some comments about what you are calculating would be helpful. (Yes, I know we have these elsewhere without comments, but we are trying to improve...)

This kind of vector algebra can generally be written using DeriVector2 is a clearer way (for the one reading the code).